Quadriga/src/main/java/edu/asu/spring/quadriga/service/passthroughproject/impl/XMLReader.java

```diff
@@ -68,14 +68,17 @@ public String getNetwork(String xml) {
      * @param tagName
      * @return The id for the given tag.
      */
-    private String getTagId(Document document, String tagName) {
+    private String getTagId(Document document, String tagName) throws DocumentParserException {
 
         Node node = document.getElementsByTagName(tagName).item(0);
         if (node == null) {
             return null;
         }
         NamedNodeMap nodeAttributeMap = node.getAttributes();
         Node idNode = nodeAttributeMap.getNamedItem("id");
+        if(idNode == null) {
+            throw new DocumentParserException("Please provide an id attribute for tag: " + tagName);
+        }
```

Comment:
Generally using exceptions to validate input is not good practice. But for now let's leave it like that.